core/src/macros.rs

```diff
@@ -18,4 +18,21 @@ macro_rules! cfg_substrate_compat {
 	};
 }
 
+macro_rules! convert_error {
+    ($module_path:ty as $delegate_ty:ty, $expr:expr) => {
+        impl From<$module_path> for $delegate_ty {
+            fn from(val: $module_path) -> Self {
+                $expr(val.into())
+            }
+        }
+    };
+    ($ty:ident $(<$( $param:ident ),+>)? as $delegate_ty:ty, $expr:expr) => {
+        impl From<$ty> for $delegate_ty {
+            fn from(val: $ty) -> Self {
+                $expr(val)
+            }
+        }
+    };
```

Comment:
This block doesn't appear to be made use of, and the `$param` that you capture isn't made use of anywhere anyways, so I'd just remove it to keep things simple :)

Reply:
done